src/App.jsx

```diff
@@ -14,33 +17,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_NAME = 'name';
+const SORT_BY_LENGTH = 'length';
+const REVERSE = 'reverse';
+let isSortParametersNull;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export function App() {
+  const goodsCopy = [...goodsFromServer];
+  const [sortType, setSortType] = useState(null);
+  const [reverseOfArray, setReverseOfArray] = useState(null);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortType === SORT_BY_NAME) {
+    goodsCopy.sort((goods1, goods2) => goods1.localeCompare(goods2));
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (sortType === SORT_BY_LENGTH) {
+    goodsCopy.sort((goods1, goods2) => goods1.length - goods2.length);
+  }
+
+  if (reverseOfArray === REVERSE) {
+    goodsCopy.reverse();
+  }
+
+  if (sortType === null && reverseOfArray === null) {
+    isSortParametersNull = null;
+  } else {
+    isSortParametersNull = 'yes';
+  }
+
+  function setResets() {
+    setSortType(null);
+    setReverseOfArray(null);
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={() => setSortType(SORT_BY_NAME)}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortType !== SORT_BY_NAME,
+          })}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          onClick={() => setSortType(SORT_BY_LENGTH)}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortType !== SORT_BY_LENGTH,
+          })}
+        >
+          Sort by length
+        </button>
+
+        {reverseOfArray === null ? (
+          <button
+            onClick={() => setReverseOfArray(REVERSE)}
+            type="button"
+            className="button is-warning is-light"
+          >
+            Reverse
+          </button>
+        ) : (
+          <button
+            onClick={() => setReverseOfArray(null)}
+            type="button"
+            className={cn('button is-warning', {
+              'is-light': reverseOfArray === null,
+            })}
+          >
+            Reverse
+          </button>
```

Comment:
```suggestion
          <button
            onClick={() => setIsReversedGoods(prev => !prev)}
            type="button"
            className={cn('button is-warning', {
              'is-light': !isReversedGoods,
            })}
          >
            Reverse
          </button>
```
If the value is boolean you can simplify the logic and don't repeat the same things for a button